service/s3/internal/customizations/update_endpoint.go

```diff
@@ -48,24 +83,66 @@ func (u *updateEndpointMiddleware) HandleSerialize(
 		return out, metadata, fmt.Errorf("unknown request type %T", req)
 	}
 
-	// Below customization only apply if bucket name is provided
-	bucket, ok := u.getBucketFromInput(in.Parameters)
-	if ok && bucket != nil {
-		if err := u.updateEndpointFromConfig(req, *bucket); err != nil {
-			return out, metadata, err
+	// check if accelerate is supported
+	if u.supportsAccelerate == nil || (u.useAccelerate && !u.supportsAccelerate(in.Parameters)) {
+		// accelerate is not supported, thus will be ignored
+		log.Println("Transfer acceleration is not supported for the operation, ignoring UseAccelerate.")
```

Comment:
This is a good example of places the SDK should consider how it exposes the concept of logging to its internals.

Reply:
yeah just followed what v1 does here. Would passing a logger on ctx be a good alternative?

Reply:
yeah we'll have to figure that out when we do the logging work :/ probably will be passed through context or additional param maybe.